jukebox/src/main/webapp/js/dbScript.js

```diff
@@ -0,0 +1,92 @@
+const NO_VIDEO = -1;
+
+var db = firebase.firestore();
+
+//Grabs the current index in queue and increments it, increment is atomic
+function incrIndex(){
+    const room_ID = getPin();
+    const increment = firebase.firestore.FieldValue.increment(1);
```

Comment:
I think you need to add this line within the update call itself - what happens if the DB write fails and we incremented a value as well? We would end up with gaps in the number of ids for that room.

Reply:
I think firebase.firestore.FieldValue.increment(1) is a function that when called on a specific field increments by a certain amount (in this case one), i think the variable name might be a bit confusing (maybe should be changed to something like incrementByOne). if the write fails, it wouldn't have queueCounter get incremented. 

For my implementation, queueCounter's number has no value associated with it (it's not tied to the actual place in the queue), as I couldn't think of an efficient way to decrement each song in the queue's QUEUE_INDEX. 

I was thinking instead we could probably sort by smallest QUEUE_INDEX and then assign a temp variable associated with its place in the queue in the parsing method.